src/modules/formBuilder/components/itemEditor/pickLists/ManagePickListOptions.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback } from 'react';
+import { useController, useFieldArray, useWatch } from 'react-hook-form';
+import PickListOption from './PickListOption';
+import ControlledSelect from '@/modules/form/components/rhf/ControlledSelect';
+import { localResolvePickList } from '@/modules/form/util/formUtil';
+import CardGroup, {
+  RemovableCard,
+} from '@/modules/formBuilder/components/itemEditor/conditionals/CardGroup';
+import { FormItemControl } from '@/modules/formBuilder/components/itemEditor/types';
+import { Component } from '@/types/gqlTypes';
+
+export interface ManagePickListOptionsProps {
+  control: FormItemControl;
+  formItemComponent: Component;
+}
+
+const ManagePickListOptions: React.FC<ManagePickListOptionsProps> = ({
+  control,
+  formItemComponent,
+}) => {
+  const pickListTypesPickList = localResolvePickList('PickListType') || [];
+
+  const { fields, append, remove, replace } = useFieldArray({
+    control,
+    name: 'pickListOptions',
+  });
+
+  const {
+    field: { onChange: onChangePickListReference },
+  } = useController({
+    control,
+    name: 'pickListReference',
+  });
+
+  // `fields` above contains the defaultValues, so we also useWatch to get recent updates
+  const fieldsWatch = useWatch({
+    control,
+    name: 'pickListOptions',
+  });
+
+  const setPickListInitialSelected = useCallback(
+    (code: string) => {
+      if (!fieldsWatch) return;
+      const newFields = fieldsWatch.map((field) => {
+        return { ...field, initialSelected: field?.code === code };
+      });
+      replace(newFields);
+    },
+    [replace, fieldsWatch]
+  );
+
+  const isCodeUnique = useCallback(
```

Comment:
Just noting this uniqueness check is not validated in the backend schema

Reply:
Thanks for flagging. I think it is sufficient to validate on the frontend.